pkg/controller/cloud/node_controller.go

```diff
@@ -340,23 +341,35 @@ func (cnc *CloudNodeController) initializeNode(ctx context.Context, node *v1.Nod
 				return err
 			}
 		}
+		return nil
+	}); err != nil {
+		utilruntime.HandleError(err)
+		return
+	}
 
-		curNode, err := cnc.kubeClient.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
+	curNode, err := cnc.kubeClient.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to get node %s: %v", node.Name, err))
+		return
+	}
 
-		cloudTaint := getCloudTaint(curNode.Spec.Taints)
-		if cloudTaint == nil {
-			// Node object received from event had the cloud taint but was outdated,
-			// the node has actually already been initialized.
-			return nil
-		}
+	cloudTaint := getCloudTaint(curNode.Spec.Taints)
+	if cloudTaint == nil {
+		// Node object received from event had the cloud taint but was outdated,
+		// the node has actually already been initialized.
+		return
+	}
 
+	type nodeModify func(*v1.Node)
+	var nodeModifyers []nodeModify
+
+	if err := func() error {
```

Comment:
This warrants AT LEAST a comment, if not being a whole new function

Reply:
Yeah, I was trying to minimize the diff for this PR. I have factored this out to keep the code readable.